mithril-signer/src/main.rs

```diff
@@ -111,11 +115,49 @@ async fn main() -> StdResult<()> {
         .with_context(|| "services initialization error")?;
 
     debug!("Started"; "run_mode" => &args.run_mode, "config" => format!("{config:?}"));
-    let mut state_machine = StateMachine::new(
+    let state_machine = StateMachine::new(
         SignerState::Init,
         Box::new(SignerRunner::new(config.clone(), services)),
         Duration::from_millis(config.run_interval),
     );
 
-    state_machine.run().await.map_err(|e| e.into())
+    let mut join_set = JoinSet::new();
```

Comment:
Why not use `select!` instead of a `JoinSet` here?

Reply:
I relied on the mechanism in place in the aggregator node. Do you think `select!` would be more suitable and that I should replace the existing code?